### PR TITLE
Multiple testsuite cleanups

### PR DESCRIPTION
**Pull request by tristantarrant**
- Consolidate junit categories
- Use surefire-junit47 everywhere so that categories work
- Move some disabled tests to the unstable group/category
- Clarify the reasons for tests being disabled
- Adds some more unstable tests (incorporates @galderz 's #5417 )

### CODE REVIEW

**Comment by tristantarrant** [comment]
Failure is unrelated.

File: atomic-factory/src/test/java/org/infinispan/atomic/sharded/collections/ShardedTreeMapTest.java [hunk 1]
```diff
@@ -31,7 +31,7 @@
     private static int NCACHES = 3;
     private static List<Cache> caches = new ArrayList<Cache>();
 
-    @Test(enabled = true)
+    @Test
```
Comment (danberindei) [inline atomic-factory/src/test/java/org/infinispan/atomic/sharded/collections/ShardedTreeMapTest.java]: Remove the annotation completely?

File: commons/pom.xml [hunk 2]
```diff
@@ -101,11 +107,11 @@
                   <Export-Package>
                      ${project.groupId}.commons.*;version=${project.version};-split-package:=error
                   </Export-Package>
-                  
+
                   <Import-Package>
                   !sun.misc,
                   *
-                  
+
```
Comment (danberindei) [inline commons/pom.xml]: This empty line looks weird, maybe it was a comment at some point and it got deleted by the release script?

File: core/src/test/java/org/infinispan/distribution/ch/CapacityFactorsFunctionalTest.java [hunk 3]
```diff
@@ -21,7 +21,7 @@
  * @author Dan Berindei
  * @since 6.0
  */
-@Test(groups = "functional", testName = "distribution.ch.CapacityFactorsFunctionalTest", enabled = false, description = "to be fixed by ISPN-6470")
+@Test(groups = "unstable", testName = "distribution.ch.CapacityFactorsFunctionalTest", description = "to be fixed by ISPN-6470")
```
Comment (danberindei) [inline core/src/test/java/org/infinispan/distribution/ch/CapacityFactorsFunctionalTest.java]: I was wondering why I haven't seen this one fail lately :)

File: parent/pom.xml [hunk 4]
```diff
@@ -1775,58 +1776,62 @@
...
             <properties>
-                <defaultTestGroup>jgroups</defaultTestGroup>
+                <defaultTestNGGroups>jgroups</defaultTestNGGroups>
             </properties>
         </profile>
         <profile>
             <id>test-transaction</id>
             <properties>
-                <defaultTestGroup>transaction</defaultTestGroup>
+                <defaultTestNGGroups>transaction</defaultTestNGGroups>
```
Comment (danberindei) [inline parent/pom.xml]: These profiles seem targeted to the core module, maybe we should move them there.

File: atomic-factory/src/test/java/org/infinispan/atomic/AtomicObjectFactoryTest.java [hunk 5]
```diff
@@ -107,7 +107,7 @@ public void distributedCacheTest() throws Exception {
 
     }
 
-    @Test(enabled = true)
+    @Test
```
[the remark this answers is not in the document]
Comment (tristantarrant) [reply atomic-factory/src/test/java/org/infinispan/atomic/AtomicObjectFactoryTest.java]: Unfortunately, when I do that, it still wants to run the test in the unstable group :( Damn TestNG and its unpredictability

[the remark this answers is not in the document]
Comment (alanfx) [reply atomic-factory/src/test/java/org/infinispan/atomic/AtomicObjectFactoryTest.java]: You could just end the sentence at "Damn TestNG"!

File: parent/pom.xml (hunk 4 above)
Comment (tristantarrant) [reply parent/pom.xml]: Let's leave them here for now, as they do no harm.

**Comment by tristantarrant** [comment]
@danberindei I addressed all of  the comments

**Comment by tristantarrant** [comment]
Updated with some additional tests, including @galderz #5417

**Comment by tristantarrant** [comment]
Green build

**Comment by galderz** [comment]
Green build!! 🎉 🎈

**Pull request merged**